lib/sass/tree/visitors/perform.rb

```diff
@@ -346,29 +346,47 @@ def visit_mixindef(node)
 
   # Runs a mixin.
   def visit_mixin(node)
+    mixin_name = nil
     @environment.stack.with_mixin(node.filename, node.line, node.name) do
-      mixin = @environment.mixin(node.name)
-      raise Sass::SyntaxError.new("Undefined mixin '#{node.name}'.") unless mixin
-
-      if node.children.any? && !mixin.has_content
-        raise Sass::SyntaxError.new(%(Mixin "#{node.name}" does not accept a content block.))
-      end
-
       args = node.args.map {|a| a.perform(@environment)}
       keywords = Sass::Util.map_vals(node.keywords) {|v| v.perform(@environment)}
       splat = self.class.perform_splat(node.splat, keywords, node.kwarg_splat, @environment)
 
+      mixin_name = node.name
+      # We let any existing mixin named `mixin` shadow dynamic includes
+      # for backwards compatability even though it will break the hell
+      # out of libraries if anyone does this.
+      mixin = @environment.mixin(mixin_name)
+
+      # The while loop is to allow any number of dynamic includes to be chained.
+      while mixin.nil? && mixin_name == "mixin"
```

Comment:
Implemented in: https://github.com/sass/sass/pull/2057/commits/c3a38aaccaeaf432f64773f23be9057077b4c5b3
Tested in:
https://github.com/sass/sass-spec/pull/864/commits/c02c59fd14e1b233c2c68de8a4d3d6a35d0ce65a